src/lib/Parser.hs

```diff
@@ -784,13 +865,27 @@ inpostfix' p op = Postfix $ do
 mkName :: String -> Name
 mkName s = Name SourceName (fromString s) 0
 
+nameToStr :: Name -> String
+nameToStr = tagToStr . nameTag
+
+-- This function is used to generate a string that is guaranteed to never shadow
+-- any user-defined name, as "#" is an invalid identifier character in normal
+-- source code.
+mkNoShadowingStr :: String -> String
+mkNoShadowingStr = (++ "#")
+
+mkInterfaceConsName :: Name -> Name
+mkInterfaceConsName =
+  GlobalName . fromString . mkNoShadowingStr . nameToStr
```

Comment:
Yeah I think that's reasonable, isn't it? Why would you ever create or unpack the dict manually? The dict is just an implementation detail really.

Reply:
I guess the only time that would happen would be if we wanted to allow people to explicitly pass typeclass dicts instead of having them be synthesized in some cases? Like you mentioned in https://github.com/google-research/dex-lang/pull/227#discussion_r486456473. But that's probably too speculative of a use case to worry about right now, so it seems fine to leave this as is.

Reply:
Oh the fact that the constructors are unreachable doesn't mean that you can't treat interface instances as values. After all the syntax can be
```
instance myInstanceName : Add Int where
  ...
```
and you can keep passing `myInstanceName` around. You'll just never be able to unpack it, but I think that's good. The only accessors you have are the instance functions.

Reply:
Ah, so you're suggesting that a user might choose between multiple overlapping instances by passing one explicitly by name, but NOT by manually constructing an instance at call time? That makes sense. I was thinking of a sort of "anonymous instance" that would be created inside a function, but I'm not sure if there's a reason to want that.